src/rgw/rgw_sal_store.h

```diff
@@ -155,36 +155,17 @@ class StoreBucket : public Bucket {
 class StoreObject : public Object {
   protected:
     RGWObjState state;
-    Bucket* bucket;
-    Attrs attrs;
```

Comment:
@dang do you know of any specific need to keep `attrs` separate from `state.attrset`? `get_attrs()`/`set_attrs()`/`has_attrs()` are all based on the latter

Reply:
No, attrs predates RGWObjState being there at all, so it can go.